### PR TITLE
BOM-1409: Upgrade to django 2.2

### PR DESCRIPTION
**Pull request by iamsobanjaved**
JIRA: [BOM-1409](https://openedx.atlassian.net/browse/BOM-1409), [BOM-1410](https://openedx.atlassian.net/browse/BOM-1410)

## Description

- Upgraded registrar to Django 2.2

### CODE REVIEW

File: requirements/base.in [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 analytics-python<2.0
 celery<4.0
-django<2.0
+django>=2.2,<3.0
```
Comment (mraarif) [inline requirements/base.in]: We shouldn't have this constraint in here

**Review by awais786 — APPROVED** [review]

File: requirements/base.in (hunk 1 above)
Comment (mraarif) [reply requirements/base.in]: maybe in a constraints file

**Review by UsamaSadiq — APPROVED** [review]

**Review by mraarif — APPROVED** [review]

File: requirements/constraints.txt [hunk 2]
```diff
@@ -22,4 +22,7 @@ mock<4.0.0
 zipp<2.0.0
 
 # Pinned until edx-drf-extension > 4.0.2 is released
-drf-jwt==1.14.0
\ No newline at end of file
+drf-jwt==1.14.0
```
Comment (kdmccormick) [inline requirements/constraints.txt]: We pin `edx-drf-extensions==4.0.3` in this PR. So, is it worth removing `drf-jwt==1.14.0` constraint?

**Review by kdmccormick — APPROVED** [review]
Approving on behalf of the Master's & MicroMaster's squad. Please let us know in #programs-devs when this merges so that we can plan a release to production strategically -- we have upcoming Master's enrollment periods, and we want to make sure we do some manual testing with Django 2.0 before rolling this out.

FYI @schenedx

File: requirements/constraints.txt (hunk 2 above)
Comment (Ayub-Khan) [reply requirements/constraints.txt]: The newer version introduces some issues. so we want to keep using 1.14.0 until those are resolved. 
https://github.com/Styria-Digital/django-rest-framework-jwt/issues/40
@iamsobanjaved kindly update the constraint comment.

**Review by Ayub-Khan — APPROVED** [review]

File: requirements/constraints.txt [hunk 3]
```diff
@@ -21,5 +21,8 @@ django-storages<1.9
 mock<4.0.0
 zipp<2.0.0
 
-# Pinned until edx-drf-extension > 4.0.2 is released
-drf-jwt==1.14.0
\ No newline at end of file
+# The newer version introduces some issues. Pinned on 1.14.0 until those are resolved.
```
Comment (Ayub-Khan) [inline requirements/constraints.txt]: it would be nice to mention BOM-1410

**Review by jmbowman — APPROVED** [review]
I agree with Ayub about linking to either the JIRA issue or upstream GitHub issue in the drf-jwt constraint comment, otherwise 👍

**Comment by iamsobanjaved** [comment]
> I agree with Ayub about linking to either the JIRA issue or upstream GitHub issue in the drf-jwt constraint comment, otherwise 👍

@jmbowman Linked JIRA issue in description

**Pull request merged**